platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/preconsensus/PreconsensusEventFile.java

```diff
@@ -201,33 +208,27 @@ public static PreconsensusEventFile of(
     public static PreconsensusEventFile of(@NonNull final Path filePath) throws IOException {
         Objects.requireNonNull(filePath, "filePath");
 
-        final boolean discontinuity;
-        if (filePath.toString().endsWith(EVENT_FILE_DISCONTINUITY_EXTENSION)) {
-            discontinuity = true;
-        } else if (filePath.toString().endsWith(EVENT_FILE_EXTENSION)) {
-            discontinuity = false;
-        } else {
+        if (!filePath.toString().endsWith(EVENT_FILE_EXTENSION)) {
             throw new IOException("File " + filePath + " has the wrong type");
         }
 
         final String fileName = filePath.getFileName().toString();
 
-        final String extension = discontinuity ? EVENT_FILE_DISCONTINUITY_EXTENSION : EVENT_FILE_EXTENSION;
-        final String[] elements =
-                fileName.substring(0, fileName.length() - extension.length()).split(EVENT_FILE_SEPARATOR);
+        final String[] elements = fileName.substring(0, fileName.length() - EVENT_FILE_EXTENSION.length())
+                .split(EVENT_FILE_SEPARATOR);
 
-        if (elements.length != 4) {
+        if (elements.length != 5) {
```

Comment:
How will we read in legacy files that have no origin?

Reply:
![thats-the-neat-part-you-dont-invincible](https://github.com/hashgraph/hedera-services/assets/56973212/e8f05386-af46-45a9-88b3-43eefeed9829)

Reply:
We currently clear the PCES on software upgrade boundaries. Look at `SwirldsPlatform.clearPCESOnSoftwareUpgradeIfConfigured()`. We won't always do this (since the PCES is required for your consensus changes), but we will do it for this particular software version.

Reply:
Once we lose the ability to clear the PCES on upgrade boundaries, file format changes will have to be performed with backwards compatibility. But I see no reason to spend that effort for this particular change since backwards compatibly is not yet needed.